pyOCD/tools/flash_tool.py

```diff
@@ -147,6 +147,8 @@ def main():
     else:
         board_selected = MbedBoard.chooseBoard(board_id=args.board_id, target_override=args.target_override,
```

Comment:
I think chooseBoard is blocking by default, so you'll need to set the 'blocking' parameter to false.